ui/ui-components/pages/app/[id]/refresh.tsx

```diff
@@ -0,0 +1,400 @@
+import Head from "next/head";
+import { useState } from "react";
+import { UseApi } from "../../../hooks/useApi";
+import { Row, Col, Form, Alert, Button, Container } from "react-bootstrap";
+import { useRouter } from "next/router";
+import PageHeader from "../../../components/PageHeader";
+import AppTabs from "../../../components/tabs/App";
+import Loader from "../../../components/Loader";
+import LoadingButton from "../../../components/LoadingButton";
+import LockedBadge from "../../../components/badges/LockedBadge";
+import SourceBadge from "../../../components/badges/SourceBadge";
+import StateBadge from "../../../components/badges/StateBadge";
+import AppRefreshQueryStats from "../../../components/app/AppRefreshQueryStats";
+import { Actions, Models } from "../../../utils/apiData";
+import { ErrorHandler } from "../../../utils/errorHandler";
+import { SuccessHandler } from "../../../utils/successHandler";
+import AppRefreshQueryScheduleList from "../../../components/app/AppRefreshSchedulesList";
+
+export default function Page(props) {
+  const {
+    errorHandler,
+    successHandler,
+  }: {
+    errorHandler: ErrorHandler;
+    successHandler: SuccessHandler;
+    types: Actions.AppOptions["types"];
+    environmentVariableOptions: Actions.AppOptions["environmentVariableOptions"];
+    optionOptions: Actions.AppOptionOptions["options"];
+  } = props;
+  const router = useRouter();
+  const { execApi } = UseApi(props, errorHandler);
+  const [app, setApp] = useState<Models.AppType>(props.app);
+  const [appRefreshQuery, setAppRefreshQuery] =
+    useState<Models.AppRefreshQueryType>(props.appRefreshQuery);
+  const [loading, setLoading] = useState(false);
+  const [editing, setEditing] = useState(false);
+  const [testLoading, setTestLoading] = useState(false);
+  const [testResult, setTestResult] = useState<{
+    success: boolean;
+    error: string;
+    message: string;
+  }>({ success: null, error: null, message: null });
+  const [ranTest, setRanTest] = useState(false);
+  const { id } = router.query;
+  const { schedules, runs, sources } = props;
+  const disabled = app?.locked !== null || appRefreshQuery?.locked !== null;
+
+  async function create(event) {
+    event.preventDefault();
+    setLoading(true);
+
+    const appRefreshQuery = { appId: app.id };
+
+    const response: Actions.AppRefreshQueryCreate = await execApi(
+      "post",
+      `/appRefreshQuery/`,
+      appRefreshQuery
+    );
+
+    if (response?.appRefreshQuery) {
+      setAppRefreshQuery(response.appRefreshQuery);
+
+      router.push(
+        "/app/[id]/refresh",
+        `/app/${response.appRefreshQuery.appId}/refresh`
```

Comment:
FYI you don't need both now, just `/app/${response.appRefreshQuery.appId}/refresh` works